modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -749,6 +749,18 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
      */
     CV_EXPORTS_W Net readNetFromModelOptimizer(const String &xml, const String &bin);
 
+    /** @brief Reads a network model stored in <a href="https://github.com/onnx/models">ONNX</a> model files.
+     *  @param onnxFile path to the .onnx file with text description of the network architecture.
+     *  @returns Network object that ready to do forward, throw an exception in failure cases.
+     */
+    CV_EXPORTS_W Net readNetFromONNX(const String &onnxFile);
+
+    /** @brief Creates 4-dimensional blob from .pb file.
```

Comment:
I think this method can read a general tensor as well but not only 4-dimensional.